app/controllers/companies_controller.rb

```diff
@@ -1,6 +1,17 @@
 class CompaniesController < ApplicationController
+  before_action :set_company, only: [:show, :edit, :update, :destroy]
+  
   def index
     @companies = Company.all
+    if params[:location]
+      @city = params[:location]
```

Comment:
no need to set this as an instance variable, we can access it via one of the companies in the view